haskell.nix/library/flake.nix

```diff
@@ -0,0 +1,105 @@
+{
+  nixConfig = {
+    flake-registry = "https://github.com/serokell/flake-registry/raw/master/flake-registry.json";
+  };
+
+  inputs = {
+    flake-utils.url = "github:numtide/flake-utils";
+    flake-compat = {
+      flake = false;
+    };
+    haskell-nix = {
+      inputs.hackage.follows = "hackage";
+      inputs.stackage.follows = "stackage";
+    };
+    hackage = {
+      flake = false;
+    };
+    stackage = {
+      flake = false;
+    };
+  };
+
+  outputs = { self, nixpkgs, haskell-nix, hackage, stackage, serokell-nix, flake-compat, flake-utils, ... }:
+    flake-utils.lib.eachDefaultSystem (system:
+      let
+        haskellPkgs = haskell-nix.legacyPackages."${system}";
+        pkgs = import nixpkgs {
+          inherit system;
+          overlays = [
+            serokell-nix.overlay
+          ];
+        };
+
+        lib = pkgs.lib;
+
+        hs-package-name = "pataq-package";
+
+        ghc-versions = [ "884" "8107" "902" ];
+
+        # invoke haskell.nix for each ghc version listed in ghc-versions
+        pkgs-per-ghc = lib.genAttrs (map (v: "ghc${v}") ghc-versions)
+          (ghc: haskellPkgs.haskell-nix.cabalProject {
+            src = haskellPkgs.haskell-nix.haskellLib.cleanGit {
+              name = hs-package-name;
+              src = ./.;
+            };
+            compiler-nix-name = ghc;
+
+            # haskell.nix configuration
+            modules = [{
+              packages.${hs-package-name} = {
+                ghcOptions = [
+                  # fail on warnings
+                  "-Werror"
+                  # disable optimisations, we don't need them if we don't package or deploy the executable
+                  "-O0"
+                ];
+              };
+
+            }];
+          });
+
+        # returns the list of all components for a package
+        get-package-components = pkg:
+          # library
+          lib.optional (pkg ? library) pkg.library
+          # haddock
+          ++ lib.optional (pkg ? library) pkg.library.haddock
+          # exes, tests and benchmarks
+          ++ lib.attrValues pkg.exes
+          ++ lib.attrValues pkg.tests
+          ++ lib.attrValues pkg.benchmarks;
+
+        # all components for each specified ghc version
+        build-all = lib.mapAttrs
+          (ghc: pkg:
+            let components = get-package-components pkg.${hs-package-name}.components;
+            in pkgs.linkFarmFromDrvs "build-all" components) pkgs-per-ghc;
+
+        # all tests for each specified ghc version
+        test-all = lib.mapAttrs
+          (ghc: pkg:
+            let tests = lib.filter builtins.isAttrs
+              (lib.attrValues pkg.${hs-package-name}.checks);
+            in pkgs.linkFarmFromDrvs "test-all" tests) pkgs-per-ghc;
+
+      in pkgs-per-ghc // {
+        # nixpkgs revision pinned by this flake
+        legacyPackages = pkgs;
+
+        # derivations that we can run from CI
+        checks = {
+
+          inherit build-all test-all;
```

Comment:
Currently `build-all` and `test-all` are sets, but not derivations, as a result `nix flake show --allow-import-from-derivation` fails with
```
error: expected a derivation
```
which is sad.

I propose to generate outputs like `checks.build-all-${ghc-version}` (or `${ghc-version}:build-all` like `haskell.nix` does) instead of set